File: benchmark/c/cpw/smallest_power_of_2_greater_than_or_equal_to_n/smallest_power_of_2_greater_than_or_equal_to_n.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



unsigned int smallest_power_of_2_greater_than_or_equal_to_n ( unsigned int n ) ;
unsigned int smallest_power_of_2_greater_than_or_equal_to_n ( unsigned int n ) {
  unsigned count = 0;
  if ( n && ! ( n & ( n - 1 ) ) ) return n;
  while ( n != 0 ) {
    n >>= 1;
    count += 1;
  }
  return 1 << count;
}

File: translation_result/else_bms/fail/c/cpw/smallest_power_of_2_greater_than_or_equal_to_n/gemini/CAPR/smallest_power_of_2_greater_than_or_equal_to_n.rs

fn smallest_power_of_2_greater_than_or_equal_to_n(mut n: u32) -> u32 {
    if n != 0 && (n & (n - 1)) == 0 {
        return n;
    }
    let mut count = 0;
    while n != 0 {
        n >>= 1;
        count += 1;
    }
    1 << count
}
